src/ipc/handler.c

```diff
@@ -839,6 +841,195 @@ static int ipc_glb_gdb_debug(uint32_t header)
+
+	return probe_deinit();
+}
+
+static inline int ipc_probe_dma_add(uint32_t header)
+{
+	struct sof_ipc_probe_dma_add_params *params = ipc_get()->comp_data;
+	int dmas_count = params->num_elems;
+
+	tracev_ipc("ipc_probe_dma_add()");
+
+	if (dmas_count > CONFIG_PROBE_DMA_MAX) {
+		trace_ipc_error("ipc_probe_dma_add() error: Invalid amount of injection DMAs specified = %d. Max is " META_QUOTE(CONFIG_PROBE_DMA_MAX) ".",
+				dmas_count);
+		return -EINVAL;
+	}
+
+	if (dmas_count <= 0) {
+		trace_ipc_error("ipc_probe_dma_add() error: Inferred amount of incjection DMAs in payload is %d. This could indicate corrupt size reported in header or invalid IPC payload.",
+				dmas_count);
+		return -EINVAL;
+	}
+
+	return probe_dma_add(dmas_count, params->probe_dma);
+}
+
+static inline int ipc_probe_dma_remove(uint32_t header)
+{
+	struct sof_ipc_probe_dma_remove_params *params = ipc_get()->comp_data;
+	int tags_count = params->num_elems;
+
+	tracev_ipc("ipc_probe_dma_remove()");
+
+	if (tags_count > CONFIG_PROBE_DMA_MAX) {
+		trace_ipc_error("ipc_probe_dma_remove() error: Invalid amount of injection DMAs specified = %d. Max is " META_QUOTE(CONFIG_PROBE_DMA_MAX) ".",
+				tags_count);
+		return -EINVAL;
+	}
+
+	if (tags_count <= 0) {
+		trace_ipc_error("ipc_probe_dma_remove() error: Inferred amount of incjection DMAs in payload is %d. This could indicate corrupt size reported in header or invalid IPC payload.",
+				tags_count);
+		return -EINVAL;
+	}
+
+	return probe_dma_remove(tags_count, params->stream_tag);
+}
+
+static inline int ipc_probe_point_add(uint32_t header)
+{
+	struct sof_ipc_probe_point_add_params *params = ipc_get()->comp_data;
+	int probes_count = params->num_elems;
+
+	tracev_ipc("ipc_probe_point_add()");
+
+	if (probes_count > CONFIG_PROBE_POINTS_MAX) {
+		trace_ipc_error("ipc_probe_point_add() error: Invalid amount of Probe Points specified = %d. Max is " META_QUOTE(CONFIG_PROBE_POINT_MAX) ".",
+				probes_count);
+		return -EINVAL;
+	}
+
+	if (probes_count <= 0) {
+		trace_ipc_error("ipc_probe_point_add() error: Inferred amount of Probe Points in payload is %d. This could indicate corrupt size reported in header or invalid IPC payload.",
+				probes_count);
+		return -EINVAL;
+	}
+
+	return probe_point_add(probes_count, params->probe_point);
+}
+
+static inline int ipc_probe_point_remove(uint32_t header)
+{
+	struct sof_ipc_probe_point_remove_params *params = ipc_get()->comp_data;
+	int probes_count = params->num_elems;
+
+	tracev_ipc("ipc_probe_point_remove()");
+
+	if (probes_count > CONFIG_PROBE_POINTS_MAX) {
+		trace_ipc_error("ipc_probe_point_remove() error: Invalid amount of Probe Points specified = %d. Max is " META_QUOTE(CONFIG_PROBE_POINT_MAX) ".",
+				probes_count);
+		return -EINVAL;
+	}
+
+	if (probes_count <= 0) {
+		trace_ipc_error("ipc_probe_point_remove() error: Inferred amount of Probe Points in payload is %d. This could indicate corrupt size reported in header or invalid IPC payload.",
```

Comment:
it's not inferred any longer but explicitly received. The trace info is also way too verbose.

Reply:
@plbossart Please consider moving this comment to follow-up implementation PR: https://github.com/thesofproject/sof/pull/1852 so that it doesn't get lost.

Reply:
it's easier to do small fixes separately, it doesn't make things simpler if you mix clarifications of trace messages with heavy duty implementation.